src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -107,14 +107,57 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 			.then(content => this.processWorkspaceRecommendations(json.parse(content.value, [])), err => []);
 	}
 
-	private processWorkspaceRecommendations(extensionsContent: IExtensionsContent): string[] {
+	private processWorkspaceRecommendations(extensionsContent: IExtensionsContent): TPromise<string[]> {
+		const regEx = new RegExp(EXTENSION_IDENTIFIER_PATTERN);
+
 		if (extensionsContent && extensionsContent.recommendations) {
-			const regEx = new RegExp(EXTENSION_IDENTIFIER_PATTERN);
-			return extensionsContent.recommendations.filter((element, position) => {
+			let filteredRecommendations = extensionsContent.recommendations.filter((element, position) => {
 				return extensionsContent.recommendations.indexOf(element) === position && regEx.test(element);
 			});
+
+			if (filteredRecommendations.length !== extensionsContent.recommendations.length) {
+				let badlyFormattedRecommendations = extensionsContent.recommendations.filter(element => {
```

Comment:
Instead of looping through `extensionsContent.recommendations` one more time here, you can use the callback for the `extensionsContent.recommendations.filter` to find the badly formatted recommendations and form the `badRecommendationsString`